lib/src/image_inlining/image_inliner.dart

```diff
@@ -169,16 +169,19 @@ Future<TransformResult> inlineImages(Asset input, ImageInliningMode mode,
       new Asset.fromString(input.id.addExtension('.map'), printer.map));
 }
 
-const _mediaTypeByExtension = const <String, String>{
+const mediaTypeByExtension = const <String, String>{
   '.jpeg': 'image/jpeg',
   '.jpg': 'image/jpeg',
   '.svg': 'image/svg+xml',
   '.png': 'image/png',
 };
 
-Future<String> encodeDataAsUri(Asset asset) async {
-  var mediaType = _mediaTypeByExtension[asset.id.extension];
-  var data = await readAll(await asset.read());
-  var encodedData = BASE64.encode(data);
+Future<String> encodeAssetAsUri(Asset asset) async {
+  return encodeMediaAsUri(mediaTypeByExtension[asset.id.extension],
+      await readAll(await asset.read()));
+}
+
+String encodeMediaAsUri(String mediaType, List<int> bytes) {
+  var encodedData = BASE64.encode(bytes);
```

Comment:
Note: there's nothing here specific about encoding media: rename as encodeBytesAsDataUri? (and rename mediaType to optional named mimeType that defaults to 'application/octet-stream'?)